src/loaders/gtdb/parse_computed_genome_attribs.py

```diff
@@ -0,0 +1,223 @@
+
+    for batch_dir in batch_dirs:
+        # retrieve checkM2 result from 'quality_report.tsv'
+        quality_file = os.path.join(result_dir, str(batch_dir), 'quality_report.tsv')
+
+        if os.path.exists(quality_file):
+            df = _read_tsv_as_df(quality_file, SELECTED_CHECKM2_FEATURES, genome_id_col='Name')
+            docs = df.apply(_checkm2_row_to_doc, args=(kbase_collection, load_ver), axis=1).to_list()
+
+            checkm2_docs.extend(docs)
+
+    return checkm2_docs
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='PROTOTYPE - Generate a JSON file for importing and upserting into ArangoDB by parsing computed '
+                    'genome attributes.')
+    required = parser.add_argument_group('required named arguments')
+    optional = parser.add_argument_group('optional arguments')
+
+    # Required flag arguments
+    required.add_argument('--tools', required=True, type=str, nargs='+',
+                          help='Extract results from tools. (e.g. gtdb_tk, checkm2, etc.)')
+    required.add_argument(f'--{loader_common_names.LOAD_VER_ARG_NAME}', required=True, type=str,
+                          help=loader_common_names.LOAD_VER_DESCR + NOTIFICATION)
+
+    # Optional arguments
+    optional.add_argument(f'--{loader_common_names.KBASE_COLLECTION_ARG_NAME}', type=str,
+                          default=loader_common_names.DEFAULT_KBASE_COLL_NAME,
+                          help=loader_common_names.KBASE_COLLECTION_DESCR + NOTIFICATION)
+    optional.add_argument('--root_dir', type=str, default=loader_common_names.ROOT_DIR,
+                          help='Root directory.')
+    optional.add_argument("-o", "--output", type=argparse.FileType('w'),
+                          default=GTDB_COMPUTED_GENOME_ATTR_FILE,
+                          help=f"output JSON file path (default: {GTDB_COMPUTED_GENOME_ATTR_FILE})")
+    args = parser.parse_args()
+
+    (tools,
+     load_ver,
+     kbase_collection,
+     root_dir) = (args.tools, getattr(args, loader_common_names.LOAD_VER_ARG_NAME),
+                  getattr(args, loader_common_names.KBASE_COLLECTION_ARG_NAME), args.root_dir)
+
+    result_dir = _locate_dir(root_dir, kbase_collection, load_ver, check_exists=True)
+
+    executed_tools = os.listdir(result_dir)
+    if set(tools) - set(executed_tools):
+        raise ValueError(f'Please ensure that all tools have been successfully executed. '
+                         f'Only the following tools have already been run: {executed_tools}')
+
+    docs = list()
+    for tool in tools:
+        try:
+            parse_ops = getattr(sys.modules[__name__], tool)
+        except AttributeError as e:
+            raise ValueError(f'Please implement parsing method for: [{tool}]') from e
+
+        docs.extend(parse_ops(root_dir, kbase_collection, load_ver))
+
+    # TODO: merge docs from different tools with the same '_key' if performance may be an issue
```

Comment:
The docs definitely need to be merged - there should be one doc per genome, regardless of the number of tools run.

Reply:
👍

Reply:
:+1: